centos-8/ks.cfg

```diff
@@ -0,0 +1,260 @@
+network --bootproto=dhcp --onboot=yes --activate
+
+# Set repo to mirror.centos.org. With CentOS 8, updates are added to BaseOS
+# and/or AppStream.  See https://wiki.centos.org/FAQ/CentOS8.
+repo --name="CentOS" --baseurl=http://mirror.centos.org/centos/$releasever/BaseOS/$basearch/os/ --cost=100
+repo --name="AppStream" --baseurl=http://mirror.centos.org/centos/$releasever/AppStream/$basearch/os/ --cost=100
+
+#set root pw here (required by KS), remove pw in post
+rootpw temp
+firewall --enabled --service=ssh
+# https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.0_release_notes/rhel-8_0_0_release#BZ-1640697
+# XXX is this still needed to get sha512?
+#authconfig --enableshadow --passalgo=sha512
+firstboot --disabled
+
+# SELinux configuration
+# By default, selinux is enforcing
+#selinux --enforcing
+selinux --permissive
+
+# Installation logging level
+logging --level=debug
+
+# System timezone
+timezone  Etc/UTC
+
+# System bootloader configuration
+bootloader --location=mbr --driveorder=vda --append="tsc=reliable divider=10 plymouth.enable=0 console=ttyS0"
+
+# Clear the Master Boot Record
+zerombr
+
+# Automatically create partitions, no LVM
+autopart --nolvm
+
+# Partition clearing information
+clearpart --all --initlabel
+
+%packages
+@core --nodefaults
+acpid
+cloud-init
+cloud-utils-growpart
+iputils
+man
+net-tools
+parted
+python3-pyserial
+vim-enhanced
+wget
+# Exclude packages that are not needed in a VM image
+-iwl*-firmware
+-alsa-*
+-btrfs-progs*
+-centos-logos
+-dracut-config-rescue
+-microcode_ctl
+-plymouth*
+
+%end
+
+%pre
+#! /bin/bash
+
+joyent_status=fail
+trap 'echo JOYENT_STATUS_PRE=$joyent_status' EXIT
+
+set -ex
+
+cd /tmp
+# -F (not -f) so that it starts logging when file appears.
+tail -F anaconda.log program.log rpm-script.log storage.log packaging.log \
+    >>/dev/ttyS0 2>&1 </dev/null &
+set +x
+joyent_status=ok
+%end
+
+%post --log=/run/ks.post.log
+#!/bin/bash
+
+joyent_status=fail
+trap 'echo JOYENT_STATUS_POST=$joyent_status' EXIT
+
+set -ex
+
+mkdir -p /run/install/joyks
+mount /dev/disk/by-label/kickstart /run/install/joyks
+
+# Install sdc-vmtools guest tools
+echo "Installing SmartOS VM Guest Tools..."
+
+sdcvmtools="/run/install/joyks/sdc-vmtools/src/linux"
+
+cp -r ${sdcvmtools}/lib/smartdc /lib/
+cp -r ${sdcvmtools}/usr/sbin/mdata-* /usr/sbin/
+cp -r ${sdcvmtools}/usr/share/man/man1/mdata-* /usr/share/man/man1/
+ln -s /usr/sbin/mdata-get /lib/smartdc/mdata-get
+echo "SmartOS VM Guest Tools installed"
+
+# Remove root password
+echo "Removing root password"
+passwd -d root
+
+## Build date used for motd and product file
+release=$(awk '$0 ~ /CentOS Linux release / { print $4 }' /etc/redhat-release |
+    cut -d. -f 1-2)
+if [[ -z "$release" ]]; then
+	echo "ks.cfg: failed to determine release" 1>&2
+	exit 1
+fi
+BUILDDATE=$(date +%Y%m%d)
+NAME="CentOS $release"
+DOCS="https://docs.joyent.com/images/linux/centos"
+
+# Create MOTD
```

Comment:
Not sure how much of stuff like this can be factored-out, but this is the third /etc/motd creation I've seen...

Reply:
(i.e. I  understand there may be sane reasons to keep it expanded in here and its friends.)

Reply:
Let's save this for a future cleanup.  Surely there are other things that could be factored out and shared across the mi-*-* repos.

Reply:
Yeah, agreed.